### PR TITLE
Project ARES: Added Tech Unlock logs

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
As the title says, you can now view who unlocked techs in the ARES Interface (if you're senior command)
I have also split the cameras in the core to an indivual network that is not shared with the rest of the ship, and placed a camera console to view this network in CIC, CO's Office, CE's Office and Brig Office

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Added AI Core camera consoles to CE, CO and Brig offices, along with CIC and Lifeboat bubble.
maptweak: Added Tech Control Console to ARES Core.
add: Added Tech Unlock logs to ARES Interface
/:cl:

### CODE REVIEW

File: maps/map_files/USS_Almayer/USS_Almayer.dmm
Comment (Nanu308) [inline maps/map_files/USS_Almayer/USS_Almayer.dmm:1]: Lgtm, consider moving the CO's console that you added cuz rn its behind the safe and that doesn't look too good.

Comment (realforest2001) [reply maps/map_files/USS_Almayer/USS_Almayer.dmm:1]: Will think of a better place to put it, though it's technically ON the safe top.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/machinery/ARES/ARES_interface.dm [hunk 1]
```diff
@@ -183,6 +183,19 @@
 		logged_orders += list(current_order)
 	data["records_requisition"] = logged_orders
 
+	var/list/logged_techs = list()
+	for(var/datum/ares_record/tech/tech_unlock as anything in datacore.records_tech)
+		if(!istype(tech_unlock))
+			continue
```
Comment (harryob) [inline code/game/machinery/ARES/ARES_interface.dm:189]: don't use as anything if you're gonna istype is anyway, that's redundant

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

File: code/modules/cm_tech/tech.dm [hunk 2]
```diff
@@ -58,6 +58,9 @@
 
 	return TRUE
 
+/datum/tech/proc/is_tier_tech()
+	return FALSE
```
Comment (Drulikar) [inline code/modules/cm_tech/tech.dm:62]: This ought to just be a var on the tech https://github.com/cmss13-devs/cmss13/blob/master/.github/guides/STYLES.md#getters-and-setters

File: code/game/machinery/ARES/ARES_interface.dm [hunk 3]
```diff
@@ -315,6 +326,9 @@
 		if("page_deleted_1to1")
 			last_menu = current_menu
 			current_menu = "deleted_talks"
+		if("page_tech")
+			last_menu = current_menu
+			current_menu = "tech_log"
 
 		// -- Delete Button -- //
 		if("delete_record")
```
Comment (Drulikar) [inline code/game/machinery/ARES/ARES_interface.dm:334]: You have not implemented deleting a record in the `switch(record.record_name)` but currently have the delete button allowed:
image

Clicking and confirming a deletion does nothing for the user for this kind of record currently.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/AresAdmin.js [hunk 4]
```diff
@@ -1573,6 +1574,120 @@ const Emergency = (props, context) => {
   );
 };
 
+const TechLogs = (props, context) => {
```
Comment (Drulikar) [inline tgui/packages/tgui/interfaces/AresAdmin.js:1577]: This page currently has no button.

File: tgui/packages/tgui/interfaces/AresAdmin.js [hunk 5]
```diff
@@ -256,6 +256,18 @@ const MainMenu = (props, context) => {
...
+              content="Tech Control Log"
+              tooltip="Review the Intel Tech Log."
+              icon="magnifying-glass-chart"
+              ml="auto"
+              px="2rem"
+              width="25vw"
+              bold
+              onClick={() => act('page_tech')}
+            />
+          </Stack.Item>
```
Comment (Drulikar) [inline tgui/packages/tgui/interfaces/AresAdmin.js:270]: This button just crashes the panel (the remote ares panel). You don't pass admin_login as data to this.
image

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Review by Drulikar — APPROVED** [review]

**Pull request merged**